js/src/gtag-events/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { addAction } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { namespace, actionPrefix } from './constants';
+import {
+	getProductObject,
+	retrievedVariation,
+	trackAddToCartEvent,
+} from './utils';
+
+/* global jQuery */
+
+addAction(
+	`${ actionPrefix }-cart-add-item`,
+	namespace,
+	( { product, quantity = 1 } ) => {
+		trackAddToCartEvent( product, quantity );
+	}
+);
+
+const addToCartClick = function ( event ) {
+	const data = event.target.dataset;
+	const product = getProductObject( { id: data.product_id } );
+
+	trackAddToCartEvent( product, data.quantity || 1 );
+};
+
+const singleAddToCartClick = function ( event ) {
+	const cartForm = event.target.closest( 'form.cart' );
+	if ( ! cartForm ) {
+		return;
+	}
+
+	const addToCart = cartForm.querySelector( '[name=add-to-cart]' );
+	if ( ! addToCart ) {
+		return;
+	}
+
+	const variationId = cartForm.querySelector( '[name=variation_id]' );
+	const quantity = cartForm.querySelector( '[name=quantity]' );
+
+	const product = getProductObject( {
+		id: parseInt( variationId ? variationId.value : addToCart.value, 10 ),
+	} );
+
+	trackAddToCartEvent(
+		product,
+		quantity ? parseInt( quantity.value, 10 ) : 1
```

Comment:
❓ So quantity can be a string? If so, shouldn't be `trackAddToCartEvent( product, data.quantity || 1 );` in line 30 handling this as well? Same goes for `variationId` case 

💅 

```
 trackAddToCartEvent(
		product,
		parseInt( quantity?.value || 1, 10) 
)
```